FloripaBus.Tests/Repositories/RoutesRepositoryTests.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+using NUnit.Framework;
+using System.Net.Http;
+using System.Net;
+using System.Text;
+
+namespace FloripaBus.Tests
+{
+	[TestFixture]
+	public class RoutesRepositoryTests
+	{
+		private const string FIND_ROUTES_URL = "https://api.appglu.com/v1/queries/findRoutesByStopName/run";
+		private const string FIND_STOPS_URL = "https://api.appglu.com/v1/queries/findStopsByRouteId/run";
+		private const string FIND_DEPARTURES_URL = "https://api.appglu.com/v1/queries/findDeparturesByRouteId/run";
+
+		private MockResponseHandler _responseHandler;
+
+		[SetUp]
+		public void SetUp()
+		{
+			_responseHandler = new MockResponseHandler();
+		}
+
+		[Test]
+		public void WhenGetsANonOkResponseShouldThrowsAnException()
+		{
+			_responseHandler.AddResponse(FIND_ROUTES_URL, new HttpResponseMessage(HttpStatusCode.BadRequest));
+			_responseHandler.AddResponse(FIND_STOPS_URL, new HttpResponseMessage(HttpStatusCode.BadRequest));
+			_responseHandler.AddResponse(FIND_DEPARTURES_URL, new HttpResponseMessage(HttpStatusCode.BadRequest));
+			var client = new HttpClient (_responseHandler);
+
+			var repository = new RouteRepository (client);
+
+			Assert.Throws<Exception> (async () => await repository.FindRoutesByStopNameAsync (string.Empty));
+			Assert.Throws<Exception> (async () => await repository.FindStopsByRouteIdAsync (0));
+			Assert.Throws<Exception> (async () => await repository.FindDeparturesByRouteIdAsync (0));
+		}
+
+		[Test]
+		public async void GivenOneRouteFromServiceThenReturnsThisRouteWithItsData()
+		{
+			var responseContent = "{ \"rows\": [ { \"id\": 32, \"shortName\": \"133\", \"longName\": \"AGRONÔMICA VIA MAURO RAMOS\", \"lastModifiedDate\": \"2012-07-23T03:00:00+0000\", \"agencyId\": 9 } ], \"rowsAffected\": 0 }";

```

Comment:
It is always a good practice to create constants to reuse the same string. The code readability is improved. Not a big deal here and on the methods below because you use them a single time, but it'd be good to think about it when having similar methods with different behaviors/outcomes.